libraries/standard/mqtt/include/mqtt.h

```diff
@@ -38,7 +38,7 @@ typedef struct MQTTContext                MQTTContext_t;
 struct MQTTTransportInterface;
 typedef struct MQTTTransportInterface     MQTTTransportInterface_t;
 
-typedef int32_t (* MQTTTransportSendFunc_t )( MQTTNetworkContext_t context,
```

Comment:
The `http_client.h` file defines `NetworkContext_t` as `typedef struct NetworkContext * NetWorkContext_t` . 
Can you update this to be consistent with HTTP?

Reply:
I don't see why we need to add a `typedef` where one's not already present, and why it needs to be `typedef`ed to a struct within the library itself. Currently we `typedef` the network context in the `mqtt_config.h` file, and if a struct is desired then it can be configured that way.

Reply:
I am fine with not explicity typedef'ng the `NetworkContext_t`. I was primarily pointing at the inconsistency between the libraries. We can make HTTP not typedef it instead for the network context.
Besides, can you add a comment to mention (for improved documentation) that the type needs to be user-defined?

Reply:
It will be addressed in a PR where we are writing common transport utility functions. We will have a common header file declaring transport interface for both HTTP and MQTT.  Documentation should be part of that PR.